From 505661225a91cbcd7a87642cd518f12d8fbf05f1 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:24 -0400
Subject: [scsi] qla2xxx: Simplify the ISPFX00 interrupt handler code for ISPFX00

Message-id: <1408391817-6389-37-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87961
O-Subject: [PATCH RHEL 7.1 e-stor 36/69] qla2xxx: Simplify the ISPFX00 interrupt handler code for ISPFX00.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Saurav Kashyap <saurav.kashyap@qlogic.com>

scsi commit id fbe9c54b1da7c1f5795bc516676544b2ced58535

Signed-off-by: Giridhar Malavali <giridhar.malavali@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_mr.c b/drivers/scsi/qla2xxx/qla_mr.c
index 919acce..023966d 100644
--- a/drivers/scsi/qla2xxx/qla_mr.c
+++ b/drivers/scsi/qla2xxx/qla_mr.c
@@ -3015,6 +3015,7 @@ qlafx00_intr_handler(int irq, void *dev_id)
  struct rsp_que *rsp;
  unsigned long flags;
  uint32_t clr_intr = 0;
+ uint32_t intr_stat = 0;
 
  rsp = (struct rsp_que *) dev_id;
  if (!rsp) {
@@ -3034,34 +3035,26 @@ qlafx00_intr_handler(int irq, void *dev_id)
  vha = pci_get_drvdata(ha->pdev);
  for (iter = 50; iter--; clr_intr = 0) {
   stat = QLAFX00_RD_INTR_REG(ha);
-  if ((stat & QLAFX00_HST_INT_STS_BITS) == 0)
+  intr_stat = stat & QLAFX00_HST_INT_STS_BITS;
+  if (!intr_stat)
    break;
 
-  switch (stat & QLAFX00_HST_INT_STS_BITS) {
-  case QLAFX00_INTR_MB_CMPLT:
-  case QLAFX00_INTR_MB_RSP_CMPLT:
-  case QLAFX00_INTR_MB_ASYNC_CMPLT:
-  case QLAFX00_INTR_ALL_CMPLT:
+  if (stat & QLAFX00_INTR_MB_CMPLT) {
    mb[0] = RD_REG_WORD(&reg->mailbox16);
    qlafx00_mbx_completion(vha, mb[0]);
    status |= MBX_INTERRUPT;
    clr_intr |= QLAFX00_INTR_MB_CMPLT;
-   break;
-  case QLAFX00_INTR_ASYNC_CMPLT:
-  case QLAFX00_INTR_RSP_ASYNC_CMPLT:
+  }
+  if (intr_stat & QLAFX00_INTR_ASYNC_CMPLT) {
    ha->aenmb[0] = RD_REG_WORD(&reg->aenmailbox0);
    qlafx00_async_event(vha);
    clr_intr |= QLAFX00_INTR_ASYNC_CMPLT;
-   break;
-  case QLAFX00_INTR_RSP_CMPLT:
+  }
+  if (intr_stat & QLAFX00_INTR_RSP_CMPLT) {
    qlafx00_process_response_queue(vha, rsp);
    clr_intr |= QLAFX00_INTR_RSP_CMPLT;
-   break;
-  default:
-   ql_dbg(ql_dbg_async, vha, 0x507a,
-       "Unrecognized interrupt type (%d).\n", stat);
-   break;
   }
+
   QLAFX00_CLR_INTR_REG(ha, clr_intr);
   QLAFX00_RD_INTR_REG(ha);
  }
diff --git a/drivers/scsi/qla2xxx/qla_mr.h b/drivers/scsi/qla2xxx/qla_mr.h
index 1e60ca2..9f0d0f4 100644
--- a/drivers/scsi/qla2xxx/qla_mr.h
+++ b/drivers/scsi/qla2xxx/qla_mr.h
@@ -336,11 +336,7 @@ struct config_info_data {
 
 #define QLAFX00_INTR_MB_CMPLT  0x1
 #define QLAFX00_INTR_RSP_CMPLT  0x2
-#define QLAFX00_INTR_MB_RSP_CMPLT 0x3
 #define QLAFX00_INTR_ASYNC_CMPLT 0x4
-#define QLAFX00_INTR_MB_ASYNC_CMPLT 0x5
-#define QLAFX00_INTR_RSP_ASYNC_CMPLT 0x6
-#define QLAFX00_INTR_ALL_CMPLT  0x7
 
 #define QLAFX00_MBA_SYSTEM_ERR  0x8002
 #define QLAFX00_MBA_TEMP_OVER  0x8005
-- 
1.7.1